src/OpenSSL/SSL.py

```diff
@@ -808,20 +809,22 @@ def set_tmp_ecdh(self, curve):
 
     def set_cipher_list(self, cipher_list):
         """
-        Change the cipher list
+        Set the list of ciphers to be used in this context.
 
-        :param cipher_list: A cipher list, see ciphers(1)
+        See the OpenSSL manual for more information (e.g.
+        :manpage:`ciphers(1)`).
+
+        :param bytes cipher_list: An OpenSSL cipher string.
         :return: None
         """
-        if isinstance(cipher_list, _text_type):
-            cipher_list = cipher_list.encode("ascii")
+        cipher_list = _text_to_bytes_and_warn("cipher_list", cipher_list)
```

Comment:
The usage is correct here but I can't help but feel like text_to_bytes_and_warn should raise a TypeError if it's not text or bytes so that we don't need the isinstance check that follows this. That should be in a separate PR though I guess since there are presumably numerous callsites for this function.